openedx/features/subscriptions/utils.py

```diff
@@ -4,18 +4,21 @@
 from student.models import User
 from enrollment import api
 from enrollment.data import get_course_enrollment
+from student.models import CourseEnrollment
 
 
-def track_subscription_enrollment(subscription_id, username, course_id):
+def track_subscription_enrollment(subscription_id, user, course_id, site):
     """
     Add user enrollment to valid subscription.
     """
     if subscription_id:
-        enrollment = api.get_enrollment(username, course_id)
-        user = User.objects.get(username=username)
-        valid_user_subscription = UserSubscription.get_valid_subscriptions(user.id).first()
+        enrollment = CourseEnrollment.objects.get(
```

Comment:
`get` without catching `DoesNotExist` exception should be avoided.

Reply:
As this get is called right after the relevant enrollment is created/updated, it should not throw an exception. Regardless, let me put it in try catch